### PR TITLE
Better solutions

### PR DESCRIPTION
**Pull request by stevenernst131**

### CODE REVIEW

File: app/supportbot/message-flow/cpu-analysis-chat/cpu-analysis-chat-flow.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    private resourceGroup;
+    private siteName;
+    private slotName;
+
+    constructor(private _appAnalysisService: AppAnalysisService, private _authService: AuthService) {
+        this._authService.getStartupInfo().subscribe(info => {
+            let pieces = info.resourceId.toLowerCase().split('/');
+            this.subscription = pieces[pieces.indexOf('subscriptions') + 1];
+            this.resourceGroup = pieces[pieces.indexOf('resourcegroups') + 1];
+            this.siteName = pieces[pieces.indexOf('sites') + 1];
```
Comment (ShekharGupta1988) [inline app/supportbot/message-flow/cpu-analysis-chat/cpu-analysis-chat-flow.ts]: maybe add this logic somewhere in common utilities class that can be utilized by anyone. I had a same functionality in one of my components (not yet checked in)

File: app/supportbot/common/graph-message/graph-message.component.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    detectorMetricsTitle: string;
+    detectorMetricsDescription: string;
+    instanceDetailTitle: string;
+    instanceDetailDescription: string;
+
+    metricsChartType: ChartType = ChartType.lineChart
+
+    constructor(private injector: Injector, private _logger: BotLoggingService, private _route: ActivatedRoute,  private _appAnalysisService: AppAnalysisService) {
+        this.detectorMetricsTitle = "Overall CPU Usage per Instance";
+        this.detectorMetricsDescription = "This graphs shows the total CPU usage on each of the instances where your application is running. " +
```
Comment (ShekharGupta1988) [inline app/supportbot/common/graph-message/graph-message.component.ts]: for this to be generic, shoudnt these descriptions be coming from an input to this component?

**Comment by ShekharGupta1988** [comment]
@stevenernst131  - Yo, I did noticed that when you do a "no" for cpu analysis, the flow seems to be broken. So that would require a fix.

File: app/supportbot/message-flow/cpu-analysis-chat/cpu-analysis-chat-flow.ts (hunk 1 above)
Comment (stevenernst131) [reply app/supportbot/message-flow/cpu-analysis-chat/cpu-analysis-chat-flow.ts]: yes I was thinking the same thing, I thought you might be adding that and I was going to use whatever you did after you checked it in

File: app/supportbot/common/graph-message/graph-message.component.ts (hunk 2 above)
Comment (stevenernst131) [reply app/supportbot/common/graph-message/graph-message.component.ts]: good catch, found the issue

Comment (stevenernst131) [reply app/supportbot/common/graph-message/graph-message.component.ts]: Yes I will make these components generic and move the inputs to the cpu detector flow service

**Pull request merged**